cajita/src/Cajita_SparseDimPartitioner.hpp

```diff
@@ -187,17 +195,73 @@ class SparseDimPartitioner : public BlockPartitioner<3>
       \brief Get the cell number in each dimension owned by the current MPI rank
       \param cart_comm MPI cartesian communicator
     */
-    std::array<int, 3> ownedCellsPerDimension( MPI_Comm cart_comm ) const
+    std::array<int, num_space_dim>
+    ownedCellsPerDimension( MPI_Comm cart_comm ) const
     {
         auto tiles_per_dim = ownedTilesPerDimension( cart_comm );
-        for ( int i = 0; i < 3; ++i )
+        for ( std::size_t d = 0; d < num_space_dim; ++d )
         {
             // compute cells_per_dim from tiles_per_dim
-            tiles_per_dim[i] <<= cell_bits_per_tile_dim * 3;
+            tiles_per_dim[d] <<= cell_bits_per_tile_dim;
         }
         return tiles_per_dim;
     }
 
+    /*!
+       \brief Get the owned number of tiles and the global tile offset of the
+       current MPI rank.
+       \param cart_comm The MPI Cartesian communicator for the partitioning.
+       \param owned_num_tile (Return) The owned number of tiles of the current
+       MPI rank in each dimension.
+       \param global_tile_offset (Return) The global tile offset of the current
+       MPI rank in each dimension
+     */
+    void
+    getOwnedTileInfo( MPI_Comm cart_comm,
+                      std::array<int, num_space_dim>& owned_num_tile,
+                      std::array<int, num_space_dim>& global_tile_offset ) const
+    {
+        // Get the Cartesian topology index of this rank.
+        std::array<int, num_space_dim> cart_rank;
+        int linear_rank;
+        MPI_Comm_rank( cart_comm, &linear_rank );
+        MPI_Cart_coords( cart_comm, linear_rank, num_space_dim,
+                         cart_rank.data() );
+
+        // Get the tiles per dimension and the remainder.
+        auto rec_mirror = Kokkos::create_mirror_view_and_copy(
+            Kokkos::HostSpace(), _rectangle_partition_dev );
+        for ( std::size_t d = 0; d < num_space_dim; ++d )
+        {
+            owned_num_tile[d] = rec_mirror( cart_rank[d] + 1, d ) -
+                                rec_mirror( cart_rank[d], d );
+            global_tile_offset[d] = rec_mirror( cart_rank[d], d );
+        }
+    }
+
+    /*!
+      \brief Get the owned number of cells and the global cell offset of the
+      current MPI rank.
+      \param cart_comm The MPI Cartesian communicator for the partitioning.
+      \param owned_num_cell (Return) The owned number of cells of the current
+      MPI rank in each dimension.
+      \param global_cell_offset (Return) The global cell offset of the current
+      MPI rank in each dimension
+    */
+    void getOwnedCellInfo(
```

Comment:
How about `ownedCellInfo` and `ownedTileInfo` - we don't return the array in these